libsigner/src/messages.rs

```diff
@@ -0,0 +1,1213 @@
+    StacksMessageCodec,
+};
+use stacks_common::util::hash::Sha512Trunc256Sum;
+use tiny_http::{
+    Method as HttpMethod, Request as HttpRequest, Response as HttpResponse, Server as HttpServer,
+};
+use wsts::common::{PolyCommitment, PublicNonce, Signature, SignatureShare};
+use wsts::curve::point::{Compressed, Point};
+use wsts::curve::scalar::Scalar;
+use wsts::net::{
+    DkgBegin, DkgEnd, DkgEndBegin, DkgPrivateBegin, DkgPrivateShares, DkgPublicShares, DkgStatus,
+    Message, NonceRequest, NonceResponse, Packet, SignatureShareRequest, SignatureShareResponse,
+};
+use wsts::schnorr::ID;
+use wsts::state_machine::signer;
+
+use crate::http::{decode_http_body, decode_http_request};
+use crate::EventError;
+
+/// Temporary placeholder for the number of slots allocated to a stacker-db writer. This will be retrieved from the stacker-db instance in the future
+/// See: https://github.com/stacks-network/stacks-blockchain/issues/3921
+/// Is equal to the number of message types
+pub const SIGNER_SLOTS_PER_USER: u32 = 11;
+
+// The slot IDS for each message type
+const DKG_BEGIN_SLOT_ID: u32 = 0;
+const DKG_PRIVATE_BEGIN_SLOT_ID: u32 = 1;
+const DKG_END_BEGIN_SLOT_ID: u32 = 2;
+const DKG_END_SLOT_ID: u32 = 3;
+const DKG_PUBLIC_SHARES_SLOT_ID: u32 = 4;
+const DKG_PRIVATE_SHARES_SLOT_ID: u32 = 5;
+const NONCE_REQUEST_SLOT_ID: u32 = 6;
+const NONCE_RESPONSE_SLOT_ID: u32 = 7;
+const SIGNATURE_SHARE_REQUEST_SLOT_ID: u32 = 8;
+const SIGNATURE_SHARE_RESPONSE_SLOT_ID: u32 = 9;
+/// The slot ID for the block response for miners to observe
+pub const BLOCK_SLOT_ID: u32 = 10;
+
+define_u8_enum!(TypePrefix {
+    BlockResponse = 0,
+    Packet = 1,
+    DkgBegin = 2,
+    DkgPrivateBegin = 3,
+    DkgEndBegin = 4,
+    DkgEnd = 5,
+    DkgPublicShares = 6,
+    DkgPrivateShares = 7,
+    NonceRequest = 8,
+    NonceResponse = 9,
+    SignatureShareRequest = 10,
+    SignatureShareResponse = 11,
+    DkgStatusSuccess = 12,
+    DkgStatusFailure = 13
+});
+
+impl TryFrom<u8> for TypePrefix {
+    type Error = CodecError;
+    fn try_from(value: u8) -> Result<Self, Self::Error> {
+        match value {
+            0 => Ok(TypePrefix::BlockResponse),
+            1 => Ok(TypePrefix::Packet),
+            2 => Ok(TypePrefix::DkgBegin),
+            3 => Ok(TypePrefix::DkgPrivateBegin),
+            4 => Ok(TypePrefix::DkgEndBegin),
+            5 => Ok(TypePrefix::DkgEnd),
+            6 => Ok(TypePrefix::DkgPublicShares),
+            7 => Ok(TypePrefix::DkgPrivateShares),
+            8 => Ok(TypePrefix::NonceRequest),
+            9 => Ok(TypePrefix::NonceResponse),
+            10 => Ok(TypePrefix::SignatureShareRequest),
+            11 => Ok(TypePrefix::SignatureShareResponse),
+            12 => Ok(TypePrefix::DkgStatusSuccess),
+            13 => Ok(TypePrefix::DkgStatusFailure),
+            _ => Err(CodecError::DeserializeError(format!(
+                "Unknown type prefix: {}",
+                value
+            ))),
+        }
+    }
+}
```

Comment:
`define_u8_enum!()` provides `from_u8(v: u8) -> Option<Self>`, so this code could just be:

```suggestion
impl TryFrom<u8> for TypePrefix {
    type Error = CodecError;
    fn try_from(value: u8) -> Result<Self, Self::Error> {
           Self::from_u8(value).ok_or_else(|| {
                CodecError::DeserializeError(format!("Unknown type prefix: {value}"))
           })
    }
}
```